vmm/src/device_manager.rs

```diff
@@ -559,13 +559,25 @@ impl Drop for ActivatedBackend {
     }
 }
 
-#[derive(Clone, Default, Serialize, Deserialize)]
+#[derive(Clone, Serialize, Deserialize)]
 struct DeviceNode {
+    id: String,
     resources: Vec<Resource>,
     parent: Option<String>,
     children: Vec<String>,
 }
 
+impl DeviceNode {
+    fn new(id: String) -> Self {
+        DeviceNode {
+            id,
+            resources: Vec::new(),
+            parent: None,
+            children: Vec::new(),
+        }
+    }
```

Comment:
No because you need the `DeviceNode` structure to implement the `Default` trait for that. And basically the Default trait would do the same thing, but also we would have to pick a default value for `id`, which is why I'd rather not implement the `Default` trait.